### PR TITLE
Remove code and tests related to deprecated 'convert' kwarg

### PR DESCRIPTION
**Pull request by ramatevish**
- [x] Added **tests** for changed code.
- [x] New features have been added to our [Hypothesis testing strategy](https://github.com/python-attrs/attrs/blob/master/tests/strategies.py).
- [x] Changes or additions to public APIs are reflected in our type stubs (files ending in ``.pyi``).
    - [x] ...and used in the stub test file `tests/typing_example.py`.
- [x] Updated **documentation** for changed code.
    - [x] New functions/classes have to be added to `docs/api.rst` by hand.
    - [x] Changes to the signature of `@attr.s()` have to be added by hand too.
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
- [x] Documentation in `.rst` files is written using [semantic newlines](http://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [x] Changes (and possible deprecations) have news fragments in [`changelog.d`](https://github.com/python-attrs/attrs/blob/master/changelog.d).

### CODE REVIEW

File: src/attr/_make.py [hunk 1]
```diff
@@ -172,26 +171,13 @@ def attrib(
     .. versionadded:: 18.1.0
        ``factory=f`` is syntactic sugar for ``default=attr.Factory(f)``.
     .. versionadded:: 18.2.0 *kw_only*
+    .. versionchanged:: 19.0.1 *convert* removed
```
Comment (ramatevish) [inline src/attr/_make.py]: I'm iffy on how this is set - does the merger update this version number at merge time?

**Comment by ramatevish** [comment]
Woof, I'm pretty sure this is failing tests because `pytest` itself uses `attrs` and is still using the depreciated kwarg.

**Comment by ramatevish** [comment]
https://github.com/pytest-dev/pytest/pull/4795

**Comment by hynek** [comment]
Nice!

I think I want to push 19.1 first with convert in place and emerge right after.

**Comment by ramatevish** [comment]
Updated now that pytest isn't breaking.

**Comment by hynek** [comment]
Thank you very much, I love the smell of deleted code in the morning!

**Pull request merged**